### PR TITLE
Composite sei

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

Adds SEI model for composite electrodes, with test for silicon/graphite

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: pybamm/models/submodels/active_material/total_active_material.py [hunk 1]
```diff
@@ -29,51 +29,25 @@ def get_coupled_variables(self, variables):
...
-        variables.update(
-            {
-                f"{Domain} electrode active material volume fraction": eps_solid,
-                f"X-averaged {domain} electrode active material volume fraction"
-                "": eps_solid_av,
-            }
-        )
+        # For each of the variables, the variable name without the phase name
+        # is constructed by summing all of the variable names with the phases
+        for variable_template in [
```
Comment (rtimms) [inline pybamm/models/submodels/active_material/total_active_material.py:34]: neat!

File: pybamm/models/submodels/interface/kinetics/total_main_kinetics.py [hunk 2]
```diff
@@ -34,39 +32,18 @@ def get_coupled_variables(self, variables):
...
-                    f"{Domain} electrode volumetric "
-                    "interfacial current density [A.m-2]": a_j_dim,
-                    f"X-averaged {domain} electrode volumetric "
-                    "interfacial current density": a_j_av,
-                }
+        # For each of the variables, the variable name without the phase name
+        # is constructed by summing all of the variable names with the phases
+        for variable_template in [
+            f"{Domain} electrode {{}}volumetric interfacial current density",
+            f"{Domain} electrode {{}}volumetric " "interfacial current density [A.m-3]",
```
Comment (rtimms) [inline pybamm/models/submodels/interface/kinetics/total_main_kinetics.py:40]: why the quotes after volumetric?

File: pybamm/models/submodels/interface/sei/sei_growth.py [hunk 3]
```diff
@@ -217,33 +213,37 @@ def set_rhs(self, variables):
             else:
                 l_cr = variables["Negative particle crack length"]
                 dl_cr = variables["Negative particle cracking rate"]
-            spreading_outer = dl_cr / l_cr * (self.param.L_outer_0 / 10000 - L_outer)
-            spreading_inner = dl_cr / l_cr * (self.param.L_inner_0 / 10000 - L_inner)
+            spreading_outer = (
+                dl_cr / l_cr * (self.phase_param.L_outer_0 / 10000 - L_outer)
```
Comment (rtimms) [inline pybamm/models/submodels/interface/sei/sei_growth.py:217]: I think we should add a comment about the `/ 10000`

**Review by rtimms — APPROVED** [review]
thanks @tinosulzer looks good!

File: pybamm/models/submodels/interface/kinetics/total_main_kinetics.py (hunk 2 above)
Comment (tinosulzer) [reply pybamm/models/submodels/interface/kinetics/total_main_kinetics.py:40]: typo :)

Comment (tinosulzer) [reply pybamm/models/submodels/interface/kinetics/total_main_kinetics.py:40]: it happens when strings need to be split over two lines and then later fit on one line

**Pull request merged**